lib/api/src/grpc/conversions.rs

```diff
@@ -1210,13 +1237,32 @@ impl From<HnswConfigDiff> for segment::types::HnswConfig {
     }
 }
 
-pub fn date_time_to_proto(date_time: NaiveDateTime) -> prost_types::Timestamp {
-    prost_types::Timestamp {
+pub fn naive_date_time_to_proto(date_time: NaiveDateTime) -> prost_wkt_types::Timestamp {
+    prost_wkt_types::Timestamp {
         seconds: date_time.timestamp(), // number of non-leap seconds since the midnight on January 1, 1970.
         nanos: date_time.nanosecond() as i32,
     }
 }
 
+pub fn date_time_to_proto(date_time: chrono::DateTime<chrono::Utc>) -> prost_wkt_types::Timestamp {
+    naive_date_time_to_proto(date_time.naive_utc())
+}
+
+pub fn date_time_from_proto(
+    date_time: prost_wkt_types::Timestamp,
+) -> chrono::DateTime<chrono::Utc> {
+    chrono::Utc.from_utc_datetime(
+        &chrono::NaiveDateTime::from_timestamp_opt(
+            date_time.seconds,
+            date_time.nanos.try_into().unwrap_or(0),
+        )
+        .unwrap_or_else(|| {
+            log::warn!("Failed to decode {date_time}, fallback to UNIX_EPOCH");
+            NaiveDateTime::UNIX_EPOCH
+        }),
```

Comment:
Instead of _silently_ failing/falling back here, I think it would be better to return and propagate an error.

Reply:
For me this seems like the only remaining thing. After which I'm very happy to approve! :heavy_check_mark:

Reply:
Good catch! Just pushed this propagation